backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSession.java

```diff
@@ -15,10 +19,18 @@ public class MolgenisSession {
   private Database database;
   private GraphQL graphqlForDatabase;
   private Map<String, GraphQL> graphqlPerSchema = new LinkedHashMap<>();
+  private static final Cache<String, GraphQL> anonymousGqlObjectCache =
+      Caffeine.newBuilder()
+          .maximumSize(100)
+          .expireAfterAccess(1, TimeUnit.MINUTES)
+          .expireAfterWrite(30, TimeUnit.MINUTES)
+          .build();
+  private GraphqlApiFactory graphqlApiFactory;
 
-  public MolgenisSession(Database database) {
```

Comment:
Maybe keep the default constructer with just the Database who calls the other constructor with a new GraphQlFactory, as you only use it in your test

Reply:
Overall, I have the idea that we should make 'user' a parameter on Database. That would make it very explicit what user is using the database and would avoid any 'login' behavior within the session and thus security issues with caching of those sessions. Obviously, that is a bit of a bigger refactor but in practice might not even be that big.